### ISSUE TITLE
Troubles with `with-context`

### ISSUE THREAD

**Issue by nimaai** [walmartlabs/lacinia#349]
Hi

I am having troubles with `with-context`. I am wrapping all my resolver functions with:

```clojure
(defn wrap-debug [resolver]
  (fn [context args value]
    (resolve/with-context (resolver context args value)
      {::test "test"})))

```
I am getting an error message: `Namespaced keywords are not supported`

When I use a simple keyword `:test` then I am getting this error message: `Field resolver returned a single value, expected a collection of values.`

All my resolvers return a scalar, a map or a collection of either of them.

More about the context of my use case see: https://github.com/leihs/leihs-borrow/blob/172f67e3e7e9e2b77a41ed2a48957f3992bb95ce/src/server/leihs/borrow/graphql/resolvers.clj#L57-L60

I am using lacinia version 0.37.0

What am I missing? Any help would be appreciated.

Matus

**Comment by hlship** [comment]
I can't think of anything in Lacinia that would generate that error message; a stack trace might help.

**Comment by hlship** [comment]
I can't remember, off the top of my head, what happens in your code snippet, when with-context is called on a ResolverResult rather than a raw value. That could be the issue.

**Comment by nimaai** [comment]
Ok, I have found out that `Namespaced keywords are not supported` error is coming from my code. I have a wrapper for each resolver which transforms the case of the `value`'s keys. So, it seems that `with-context` temporarily stores the context extension map in the resolved `value` before merging it into the application context. That was causing the problem in context of my code.

I still haven't figured out where was `Field resolver returned a single value, expected a collection of values.` coming from, but by reordering my resolver wrappers' chain it got solved by itself too. As you have pointed out, some `ResolverResult` in between perhaps.

I guess this issue can be closed. Thanks.
